hadoop-plugin/src/main/groovy/com/linkedin/gradle/dependency/CheckDependencyTask.groovy

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2015 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.linkedin.gradle.dependency;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.Dependency;
+import org.gradle.api.tasks.TaskAction;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.Unmarshaller;
+
+class CheckDependencyTask extends DefaultTask {
+
+  Map<Dependency, List<DependencyPattern>> matchedDependencyPatterns;
+  String DEPENDENCY_PATTERN_FILE = "sample-dependency-pattern.xml";
+  boolean errorFound = false;
+  boolean warningFound = false;
+
+
+  @TaskAction
+  void checkDependencies() {
+
+    matchedDependencyPatterns = new HashMap<Dependency, List<DependencyPattern>>();
+
+    // get all the dependencies used in the project
+    Set<Dependency> dependencies = getDependencies(project);
+    project.logger.debug("Dependencies found : ${dependencies.toString()}")
+
+    // get the List of DependencyPattern against which the dependencies should be checked
+    List<DependencyPattern> dependencyPatternList = getDependencyPatterns(project).getDependencyPattern();
+
+    project.logger.debug("DependencyPatterns found : ${dependencyPatternList.toString()}")
+
+    fillMatchedDependencyPatterns(dependencies,dependencyPatternList);
```

Comment:
This method can return the Map and we can have it as a local variable instead of having the Map as a field